### PR TITLE
Add User relations to users/me

### PR DESCRIPTION
**Pull request by a13z**
Add 'artworks', 'collected', 'memoirs', 'interviews', 'orders', 'links' relations to users/me endpoint

### CODE REVIEW

**Comment by MattieBelt** [comment]
Hey Alfonso, I would not extend the `fetchAuthenticatedUser` as this will populate the whole user object in the state (so on every request).

> Performance was the main reason

([source](https://forum.strapi.io/t/get-currrent-logged-user-with-custom-data-added-to-the-user-collection-type/1024/5))

Populating in the `me` function would be more performant for the whole application.
You can implement something like this:  

```js
  /**
   * Retrieve authenticated user.
   * @return {Object|Array}
   */
  async me(ctx) {
    let user = ctx.state.user;

    if (!user) {
      return ctx.badRequest(null, [{ messages: [{ id: 'No authorization header was found' }] }]);
    }

   user = strapi.plugins['users-permissions'].services.user.fetch({ id: user.id }, ['artworks', 'collected', 'memoirs', 'interviews', 'orders', 'links'])
   

    ctx.body = sanitizeUser(user);
  },
```

If there will be more places (e.g. policies and controllers) that need more user context then editing the  `fetchAuthenticatedUser` would be best, but in this case you are populating (potentially) big relations.

**Comment by Kyle772** [comment]
Hey guys! I was thinking about this last night after I had stepped away. We could pull out all the user data into a new route "/user/mydata" or "/user/me/data/" or something similar. 

@MattieBelt Good eye! Alfonso and I actually looked at that exact thread yesterday. I think in that thread they are talking about "/user/me" specifically which I believe has some performance implications of its own (correct me if I'm wrong there). I think this commit was Alfonso trying to take that thread into consideration by adding the extra data to the auth request for consistency's sake between "/user/:id" and "/user/me" despite the performance implications.

I think in the end we should try to keep with the original ethos of the strapi devs here and just make it so you have to intentionally request extra data through another endpoint. This way we can keep to the strapi dev's intentional decision, keep performance in mind, and make the extra data easily accessible despite those points. 

This would also be a good opportunity to expand on the content policies (ie ["isMe"]: isauthenticated, shares the ctx.id, and doesn't pull too much data to perform that policy check)

Thoughts?

**Comment by a13z** [comment]
Thank you for the comments @MattieBelt and @Kyle772 
I think extending `me` in the controller is the most reasonable thing to do in this case. That's what I wanted to do but I found this post https://forum.strapi.io/t/get-currrent-logged-user-with-custom-data-added-to-the-user-collection-type/1024 which is what I ended up doing.

I'll apply that change and submit the code

File: controllers/User.js [hunk 1]
```diff
@@ -145,6 +145,8 @@ module.exports = {
       return ctx.badRequest(null, [{ messages: [{ id: 'No authorization header was found' }] }]);
     }
 
+    user = strapi.plugins['users-permissions'].services.user.fetch({ id: user.id }, ['artworks', 'collected', 'memoirs', 'interviews', 'orders', 'links'])
```
Comment (MattieBelt) [inline controllers/User.js:148]: This will throw an error, user L142 needs to be an `let` instead of `const`

**Review by MattieBelt — CHANGES_REQUESTED** [review]

File: controllers/User.js (hunk 1 above)
Comment (a13z) [reply controllers/User.js:148]: you have really good eyes! 
It is changed, thank you!

**Comment by MattieBelt** [comment]
Hey @Kyle772,

> Looked at that exact thread yesterday. I think in that thread they are talking about "/user/me" specifically which I believe has some performance implications of its own (correct me if I'm wrong there).

Yes this is mainly to extend the `user/me` endpoint, to populate custom relations. Not sure why Derrick recommended it this way. 

> I think in the end we should try to keep with the original ethos of the strapi devs here and just make it so you have to intentionally request extra data through another endpoint. This way we can keep to the strapi dev's intentional decision, keep performance in mind, and make the extra data easily accessible despite those points.

I do think that leaving as much as the original code intact is a good idea, but the plugin is already forked, so it wouldn't give an  specific conflicts when having updates. So for the user to retrieve it's own data, which is where `/users/me` is designed for is it fine/recommended to do it this way. 


I have asked Derricks directly, but it's good for now.

**Review by MattieBelt — APPROVED** [review]
@a13z LGTM

**Pull request merged**